paddle/fluid/operators/listen_and_serv_op.cc

```diff
@@ -275,14 +275,17 @@ void ListenAndServOp::RunImpl(const framework::Scope &scope,
   request_checkpoint_handler_.reset(new distributed::RequestCheckpointHandler(
       sync_mode, checkpoint_block_id));
 
+  int thread_pool_size =
+      static_cast<int>(framework::ThreadPool::GetInstance()->Threads());
   rpc_service_->RegisterRPC(distributed::kRequestSend,
-                            request_send_handler_.get());
+                            request_send_handler_.get(), thread_pool_size * 2);
```

Comment:
Maybe create only one list of threads inside `rpc_service_` is enough? in `grpc_server.cc:285`